chasm/tree.go

```diff
@@ -727,8 +771,12 @@ func (n *Node) syncSubField(fieldV reflect.Value, fieldN string, nodePath []stri
 		// Field is not empty but tree node is not set. It means this is a new field, and a node must be created.
 		childNode := newNode(n.nodeBase, n, fieldN)
 
-		if err = assertStructPointer(reflect.TypeOf(internal.value())); err != nil {
-			return
+		switch internal.value().(type) {
+		case []string: // Pointer
+		default: // Component | Data
+			if err = assertStructPointer(reflect.TypeOf(internal.value())); err != nil {
```

Comment:
I think it is the case based on my reading of the code but want to double check: If Component or Data field type, it (`T`) needs to be pointer to a struct. For Pointer field type, `T` can be an interface.

Reply:
Yes, the answer is "yes" :-).

In 
```go
func ComponentPointerTo[C Component](
	ctx MutableContext,
	c C,
) (Field[C], error) {
```
`C` can be and interface, but `c` will be always an instance of a concrete type which should match the type of what it points to. If they are mismatch, it can be hard to debug, I guess.